pkg/steps/cluster_claim.go

```diff
@@ -21,15 +21,17 @@ import (
 	"github.com/openshift/ci-tools/pkg/results"
 	"github.com/openshift/ci-tools/pkg/steps/loggingclient"
 	"github.com/openshift/ci-tools/pkg/util"
+	"github.com/openshift/ci-tools/pkg/util/watchingclient"
 )
 
 type clusterClaimStep struct {
 	as           string
 	clusterClaim *api.ClusterClaim
-	hiveClient   ctrlruntimeclient.Client
+	hiveClient   watchingclient.Client
 	client       loggingclient.LoggingClient
 	jobSpec      *api.JobSpec
 	wrapped      api.Step
+	waitForClaim func(client watchingclient.Client, ns, name string, timeout time.Duration) error
```

Comment:
I do not understand why this is a mutable field on the step. Please explain?

Reply:
Originally, I need to override the function for testing `run` function of the step.
But now we test `acquireCluster` instead and the function `waitForClaim ` is passed as an input.

`waitForClaim` becomes now a hard-coded function in `run`.